### PR TITLE
Feature/23263 rdbms search client process definition

### LINKED ISSUE #23263: Implement Exporter and SearchClient for ProcessDefinition
The RDBMS Camunda 8 REST API V2 should support querying for ProcessDefinitions.

**AcceptanceCriteria:**
- the RDBMS module fully implements the ProcessDefinitionSearchClient with all filters and fields
- the RDBMS module can process all relevant Zeebe records for the ProcessDefinition

### PR DESCRIPTION
**Pull request by stoerti**
## Description

Adds support to search for processDefinitions via SearchAPI.

closes #23263

### CODE REVIEW

File: db/rdbms/src/main/resources/db/changelog/rdbms-support/changesets/create_engine.xml [hunk 1]
```diff
@@ -23,10 +23,16 @@
       </column>
       <column name="PROCESS_DEFINITION_ID" type="VARCHAR(200)" />
       <column name="NAME" type="VARCHAR(200)" />
+      <column name="RESOURCE_NAME" type="VARCHAR(400)" />
```
Comment (p-wunderlich) [inline db/rdbms/src/main/resources/db/changelog/rdbms-support/changesets/create_engine.xml:26]: Hm in general we should maybe sync the size of the varchars with c7. For Resource_Name, there they use 4000 .... 

```
create table ACT_RE_PROCDEF (
    ID_ nvarchar(64) not null,
    REV_ int,
    CATEGORY_ nvarchar(255),
    NAME_ nvarchar(255),
    KEY_ nvarchar(255) not null,
    VERSION_ int not null,
    DEPLOYMENT_ID_ nvarchar(64),
    RESOURCE_NAME_ nvarchar(4000),
    DGRM_RESOURCE_NAME_ nvarchar(4000),
    HAS_START_FORM_KEY_ tinyint,
    SUSPENSION_STATE_ tinyint,
    TENANT_ID_ nvarchar(64),
    VERSION_TAG_ nvarchar(64),
    HISTORY_TTL_ int,
    STARTABLE_ bit NOT NULL default 1,
    primary key (ID_)
);
```

**Review by p-wunderlich — APPROVED** [review]

File: db/rdbms/src/main/java/io/camunda/db/rdbms/read/service/ProcessDefinitionReader.java [hunk 2]
```diff
@@ -7,19 +7,42 @@
...
+  }
+
+  public SearchResult search(final ProcessDefinitionDbQuery filter) {
+    LOG.trace("[RDBMS DB] Search for process instance with filter {}", filter);
+    final var totalHits = processDefinitionMapper.count(filter);
+    final var hits = processDefinitionMapper.search(filter);
+    return new SearchResult(hits, totalHits);
   }
+
+  public record SearchResult(List<ProcessDefinitionEntity> hits, Integer total) {}
```
Comment (danielkelemen) [inline db/rdbms/src/main/java/io/camunda/db/rdbms/read/service/ProcessDefinitionReader.java:47]:
```suggestion
  public record SearchResult(List<ProcessDefinitionEntity> hits, Long total) {}
```

File: zeebe/exporters/rdbms-exporter/src/main/java/io/camunda/exporter/rdbms/ProcessExportHandler.java [hunk 3]
```diff
@@ -29,32 +32,39 @@ public ProcessExportHandler(final ProcessDefinitionWriter processDefinitionWrite
 
   @Override
   public boolean canExport(final Record<Process> record) {
-    return record.getIntent() == ProcessIntent.CREATED;
+    // do not react on ProcessEvent.DELETED to keep historic data
```
Comment (danielkelemen) [inline zeebe/exporters/rdbms-exporter/src/main/java/io/camunda/exporter/rdbms/ProcessExportHandler.java:35]: 💬 Should the entity be marked somehow maybe when it's deleted?

**Review by danielkelemen — CHANGES_REQUESTED** [review]
Added a few comments, looks good overall.
Nice leet tests :)

File: zeebe/exporters/rdbms-exporter/src/main/java/io/camunda/exporter/rdbms/ProcessExportHandler.java (hunk 3 above)
Comment (stoerti) [reply zeebe/exporters/rdbms-exporter/src/main/java/io/camunda/exporter/rdbms/ProcessExportHandler.java:35]: So far neither the existing exporter does this nor does the REST Api have a field for this. But I will clarify and create a follow-up ticket if needed.

**Review by danielkelemen — APPROVED** [review]
👍 Looks good!

**Pull request merged**